server/src/linter.ts

```diff
@@ -63,8 +63,10 @@ export default class Linter {
     folders: LSP.WorkspaceFolder[],
   ): Promise<ShellcheckResult> {
     const args = ['--format=json1', '--external-sources', `--source-path=${this.cwd}`]
-    for (const folder of folders) {
-      args.push(`--source-path=${folder.name}`)
+    if (Array.isArray(folders)) {
```

Comment:
In which case is `folders` not an array? What is it?

I'm asking as the type signature here explicitly says that it should always be an array of `LSP.WorkspaceFolder`

Reply:
according to rough looking to lsp specification, 
looks it said `It can be 'null' if the client supports workspace folders but none are configured`.

Reply:
That case is handled here: https://github.com/bash-lsp/bash-language-server/blob/main/server/src/server.ts#L92

Reply:
the problem looks was the previous statement before L92.